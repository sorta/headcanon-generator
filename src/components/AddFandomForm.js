import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AddFandomForm extends Component {
  static propTypes = {
    addFandom: PropTypes.func,
  };

  static defaultProps = {
    addFandom: () => {},
  };

  nameRef = React.createRef();

  createFandom = (ev) => {
    ev.preventDefault();
    const fandom = {
      name: this.nameRef.current.value
    };
    this.props.addFandom(fandom);
    ev.currentTarget.reset();
  }

  render() {
    return (
      <form className="AddFandomForm" onSubmit={this.createFandom}>
        <input name="name" ref={this.nameRef} type="text" placeholder="Name" required />
        <button type="submit">+</button>
      </form>
    );
  }
}

export default AddFandomForm;
