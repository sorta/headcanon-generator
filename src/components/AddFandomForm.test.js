import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import AddFandomForm from './AddFandomForm';

const props = {
  addFandom: jest.fn(),
  isManaging: () => false,
};

// Tests for regular mode
describe("AddFandomForm", () => {
  const component = shallow(<AddFandomForm { ...props } />);

  it('renders and matches our snapshot', () => {
    const snapshot = renderer.create(
      <AddFandomForm { ...props } />
    );
    const tree = snapshot.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be null when not managing', () => {
    expect(component.getElement()).toBe(null);
  });
});


// Tests for managing mode
props.isManaging = () => true;
describe("AddFandomForm in managing mode", () => {

  it('renders and matches our snapshot', () => {
    const snapshot = renderer.create(
      <AddFandomForm { ...props } />
    );
    const tree = snapshot.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have a name input field', () => {
    const component = shallow(<AddFandomForm { ...props } />);
    expect(component.find('input[name="name"]').exists()).toBe(true);
  });

  it('should have a submit button', () => {
    const component = shallow(<AddFandomForm { ...props } />);
    expect(component.find('button[type="submit"]').exists()).toBe(true);
  });

  it('form submit should call addFandom', () => {
    const component = mount(<AddFandomForm { ...props } />);
    component.find('input[name="name"]').instance().value = 'TESTING';
    component.find('button[type="submit"]').simulate('submit');
    expect(props.addFandom.mock.calls.length).toEqual(1);
  });
});
