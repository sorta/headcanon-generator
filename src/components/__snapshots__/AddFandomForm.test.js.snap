// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFandomForm in managing mode renders and matches our snapshot 1`] = `
<form
  className="AddFandomForm form-add control-row"
  onSubmit={[Function]}
>
  <input
    className="control-stretch"
    name="name"
    placeholder="New Fandom Name"
    required={true}
    type="text"
  />
  <button
    type="submit"
  >
    +
  </button>
</form>
`;

exports[`AddFandomForm renders and matches our snapshot 1`] = `
<form
  className="AddFandomForm form-add control-row"
  onSubmit={[Function]}
>
  <input
    className="control-stretch"
    name="name"
    placeholder="New Fandom Name"
    required={true}
    type="text"
  />
  <button
    type="submit"
  >
    +
  </button>
</form>
`;
